### DISCUSSION TITLE
Optimum size for CFG_TUD_CDC_EP_BUFSIZE?

### DISCUSSION THREAD

**Discussion by alanbowman** [hathach/tinyusb#1442] [Q&A]
I'm trying to use USB-CDC to get data from a RP2040, generating new data packets every 30us.  Ideally I'd like each packet to be 32 bytes in size, although they can be smaller.  I realise that's a tall order for USB speeds, but I'm trying to get as close as possible.  (I don't need the data to arrive at the host every 30us, that's just how frequently it's being generated on the RP2040.)

In the example tusb_config.h file, there's the following:
```c
// CDC Endpoint transfer buffer size, more is faster
#define CFG_TUD_CDC_EP_BUFSIZE (64)
```
Given that I have RAM to spare, how much bigger gets me how much faster?  Is there a point beyond which it makes no sense to keep increasing?  (It does appear to get better the larger I make that number, up to 2^15, but documentation suggests that much lower numbers than that would be more normal).  I've also played with increasing the CDC FIFO buffer sizes in line with the EP buffer size, although I haven't determined the best relationship between the two.

**Comment by hathach** [comment]
`EP_BUFSIZE` will reduce the overhead from controller-> usbd tud_task() cycles and allow to gain more bandwidth. Otherwise the hw must wait for tud_task() to put data to fifo before continuing. If you have SRAM to spare, try to increase it to 512 or more, it will definitely helps by how much is totally depends on your application
